pkg/api/v20220201/openshiftcluster.go

```diff
@@ -0,0 +1,277 @@
+}
+
+// ProvisioningState represents a provisioning state.
+type ProvisioningState string
+
+// ProvisioningState constants.
+const (
+	ProvisioningStateCreating      ProvisioningState = "Creating"
+	ProvisioningStateUpdating      ProvisioningState = "Updating"
+	ProvisioningStateAdminUpdating ProvisioningState = "AdminUpdating"
+	ProvisioningStateDeleting      ProvisioningState = "Deleting"
+	ProvisioningStateSucceeded     ProvisioningState = "Succeeded"
+	ProvisioningStateFailed        ProvisioningState = "Failed"
+)
+
+// ClusterProfile represents a cluster profile.
+type ClusterProfile struct {
+	// The pull secret for the cluster.
+	PullSecret string `json:"pullSecret,omitempty"`
+
+	// The domain for the cluster.
+	Domain string `json:"domain,omitempty"`
+
+	// The version of the cluster.
+	Version string `json:"version,omitempty"`
+
+	// The ID of the cluster resource group.
+	ResourceGroupID string `json:"resourceGroupId,omitempty"`
+}
+
+// ConsoleProfile represents a console profile.
+type ConsoleProfile struct {
+	// The URL to access the cluster console.
+	URL string `json:"url,omitempty"`
+}
+
+// ServicePrincipalProfile represents a service principal profile.
+type ServicePrincipalProfile struct {
+	// The client ID used for the cluster.
+	ClientID string `json:"clientId,omitempty" mutable:"true"`
+
+	// The client secret used for the cluster.
+	ClientSecret string `json:"clientSecret,omitempty" mutable:"true"`
+}
+
+// SoftwareDefinedNetwork constants.
+type SoftwareDefinedNetwork string
+
+const (
+	SoftwareDefinedNetworkOVNKubernetes SoftwareDefinedNetwork = "OVNKubernetes"
+	SoftwareDefinedNetworkOpenShiftSDN  SoftwareDefinedNetwork = "OpenShiftSDN"
+)
+
+// NetworkProfile represents a network profile.
+type NetworkProfile struct {
+	// The software defined network (SDN) to use when installing the cluster.
+	SoftwareDefinedNetwork SoftwareDefinedNetwork `json:"softwareDefinedNetwork,omitempty"`
+
+	// The CIDR used for OpenShift/Kubernetes Pods.
+	PodCIDR string `json:"podCidr,omitempty"`
+
+	// The CIDR used for OpenShift/Kubernetes Services.
+	ServiceCIDR string `json:"serviceCidr,omitempty"`
+}
+
+// EncryptionAtHost represents encryption at host state
+type EncryptionAtHost string
+
+// EncryptionAtHost constants
+const (
+	EncryptionAtHostEnabled  EncryptionAtHost = "Enabled"
+	EncryptionAtHostDisabled EncryptionAtHost = "Disabled"
+)
+
+// MasterProfile represents a master profile.
+type MasterProfile struct {
+	// The size of the master VMs.
+	VMSize VMSize `json:"vmSize,omitempty"`
+
+	// The Azure resource ID of the master subnet.
+	SubnetID string `json:"subnetId,omitempty"`
+
+	// Whether master virtual machines are encrypted at host.
+	EncryptionAtHost EncryptionAtHost `json:"encryptionAtHost,omitempty"`
+
+	// The resource ID of an associated DiskEncryptionSet, if applicable.
+	DiskEncryptionSetID string `json:"diskEncryptionSetId,omitempty"`
+}
+
+// VMSize represents a VM size.
+type VMSize string
```

Comment:
I think for L-series and additional vmsize's moving forward we may need to do some consideration on this moving forward. I'm a bit rusty coming back from PTO, but something to do with adding in new VM types changes the allowed responses which is considered a breaking API change. Just thinking out loud here.